### PR TITLE
GDB-8022: add keyboard shortcut dialog

### PR DESCRIPTION
**Pull request by boyan-tonchev**
## What
Added button in right bottom corner of YASQE. When button is clicked a dialog is opened with description of all keyboard shortcuts which "ontotext-yasgui-web-component" supports.

## Why
"ontotext-yasgui-web-component" supports some keyboard shortcuts, and we have to inform user of component how to use them.

## How
- Implement a "keyboard-shortcuts-dialog" component.
- Implement abstraction about creation of keyboard shortcuts description into dialog and registration of shortcuts.

### CODE REVIEW

**Comment by boyan-tonchev** [comment]
Jenkins retry

File: ontotext-yasgui-web-component/src/services/yasgui/configuration/yasgui-configuration-builder.ts [hunk 1]
```diff
@@ -102,6 +107,27 @@ export class YasguiConfigurationBuilder {
     return config;
   }
 
+  private initShortcuts(config: YasguiConfiguration): void {
+    const keyboardShortcutDescriptions = KeyboardShortcutService.initKeyboardShortcutMapping();
+    const extraKeys = {};
+    keyboardShortcutDescriptions.forEach(keyboardShortcutDescription => {
+      keyboardShortcutDescription.keyboardShortcuts.forEach(keyboardShortcut => {
```
Comment (svilenvelikov) [inline ontotext-yasgui-web-component/src/services/yasgui/configuration/yasgui-configuration-builder.ts:114]: Replace these nested loops with map->forEach or map->map

File: ontotext-yasgui-web-component/src/services/yasgui/configuration/yasgui-configuration-builder.ts [hunk 2]
```diff
@@ -102,6 +107,27 @@ export class YasguiConfigurationBuilder {
...
+        extraKeys[keyboardShortcut] = keyboardShortcutDescription.executeFunction;
+      });
+    });
+    config.yasguiConfig.yasqe.extraKeys = extraKeys;
+
+    config.yasguiConfig.yasqe.keyboardShortcutDescriptions = keyboardShortcutDescriptions.map(keyboardShortcutDescription => keyboardShortcutDescription.NAME.toString());
```
Comment (svilenvelikov) [inline ontotext-yasgui-web-component/src/services/yasgui/configuration/yasgui-configuration-builder.ts:120]: Maybe use toLowerCase or even change the enum KeyboardShortcutName values to be lowercase

**Review by svilenvelikov — CHANGES_REQUESTED** [review]

File: ontotext-yasgui-web-component/src/components/ontotext-dialog-web-component/ontotext-dialog-web-component.scss [hunk 3]
```diff
@@ -1,7 +1,8 @@
+@import 'src/css/variables';
 @import 'src/css/common';
 
 :host {
-  display: block;
+  //display: block;
```
Comment (svilenvelikov) [inline ontotext-yasgui-web-component/src/components/ontotext-dialog-web-component/ontotext-dialog-web-component.scss:5]: delete it

File: ontotext-yasgui-web-component/src/services/yasgui/configuration/yasgui-configuration-builder.ts [hunk 4]
```diff
@@ -109,13 +109,13 @@ export class YasguiConfigurationBuilder {
...
   private initShortcuts(config: YasguiConfiguration): void {
     const keyboardShortcutDescriptions = KeyboardShortcutService.initKeyboardShortcutMapping();
-    const extraKeys = {};
-    keyboardShortcutDescriptions.forEach(keyboardShortcutDescription => {
-      keyboardShortcutDescription.keyboardShortcuts.forEach(keyboardShortcut => {
-        extraKeys[keyboardShortcut] = keyboardShortcutDescription.executeFunction;
-      });
-    });
-    config.yasguiConfig.yasqe.extraKeys = extraKeys;
+    config.yasguiConfig.yasqe.extraKeys = keyboardShortcutDescriptions.reduce((extra, keyboardShortcutDescription) => {
```
Comment (svilenvelikov) [inline ontotext-yasgui-web-component/src/services/yasgui/configuration/yasgui-configuration-builder.ts:112]: Again, you can can avoid this nesting and complexity by flattening this by just mapping the keyboardShortcutDescriptions and then either reduce or loop-assign as in your first implementation.

**Comment by boyan-tonchev** [comment]
Jenkins retry

**Comment by boyan-tonchev** [comment]
Jenkins retry

**Comment by boyan-tonchev** [comment]
Jenkins rebuild

**Comment by boyan-tonchev** [comment]
Jenkins retry

**Comment by boyan-tonchev** [comment]
Jenkins retry

**Comment by boyan-tonchev** [comment]
Jenkins retry

**Pull request merged**